tests/suites/test_suite_mpi.function

```diff
@@ -763,6 +763,271 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void mpi_core_cond_assign( data_t * input_X,
+                           data_t * input_Y )
+{
+    #define MAX_LEN 64
+    mbedtls_mpi_uint *X = NULL;
+    mbedtls_mpi_uint *Y = NULL;
+    size_t len_X = CHARS_TO_LIMBS( input_X->len );
+    size_t len_Y = CHARS_TO_LIMBS( input_Y->len );
```

Comment:
X and Y should have the same length. The test code accepts different lengths, but it basically truncates X to `len_Y`, and it causes a buffer overflow if Y is the larger one. Instead the test should have a single length value, and should assert that X and Y have the same limb size (or the stronger condition that `input_X` and `input_Y` have the same length).

Goes for both swap and assign, for both core and mod.